cmd/kubeadm/app/constants/constants.go

```diff
@@ -361,7 +361,7 @@ var (
 	SupportedEtcdVersion = map[uint8]string{
 		10: "3.1.12",
 		11: "3.2.18",
-		12: "3.2.24",
+		12: "3.2.24-1",
```

Comment:
this gets parsed as a semantic version in `EtcdSupportedVersion()`, which happens to work for now, but it makes me a bit nervous. (I'm not sure why we even parse it though, since it basically gets ignored otherwise).

Reply:
@ixdy 
it just treats it the `-1` as preRelease "extra" instead of it's purpose for being a revision.
this is stable and it would work, but is obviously wrong as SupportedEtcdVersion is now a list of tags and not versions. :\

it needs modifications that are kind of too late to do in 1.12.
so i'd say we either merge as is or wait for 1.13.

Reply:
right, this happens to work. we could maybe submit this PR as-is for 1.12, and then fix the names in 1.13?

Reply:
just verified and the `3.2.24` manifest lists seems to work, which would suggests that kubeadm should work as well. will try that later on an actual cluster, but probably best to leave this for after 1.12 and also include the name -> tag change.

although, i'm not exactly sure how to proceed with that change TBH.
the semantic parsing still has to happen in one form or another over the tag.